org.jacoco.core/src/org/jacoco/core/internal/flow/Instruction.java

```diff
@@ -25,10 +27,13 @@
 
 	private int branches;
 
-	private int coveredBranches;
+	// TODO in majority of cases BitSet not needed, needed only for big switches
```

Comment:
@Godin Agreed. We can do some performance testing later and see where the actual bottlenecks are.